### PR TITLE
Lint and re-structure documentation

### PR DESCRIPTION
**Pull request by zanetworker**
**What this PR does / why we need it**:

This pull request restructures documentation under the "docs" package
and restructure the yaml files based on their intent (e.g., under
machine_classes, secrets, or machines_sets_deployments)

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Restructure documentation so they are easier to consume. 
```

### CODE REVIEW

File: docs/usage/machine.md [hunk 1]
```diff
@@ -1,41 +1,62 @@
...
 ## Setting up your usage environment
 
 * Follow the [steps described here](prerequisite.md)
 
-## Important :warning: 
-- Make sure that the `kubernetes/machine.yaml` points to the same class name as the `kubernetes/aws-machine-class.yaml`.
-- Similarily `kubernetes/aws-machine-class.yaml` secret name and namespace should be same as that mentioned in `kubernetes/aws-secret.yaml`
+## Important :warning
+
+> Make sure that the `kubernetes/machines_sets_deployments/machine.yaml` points to the same class name as the `kubernetes/machine_classes/aws-machine-class.yaml`.
```
Comment (prashanth26) [inline docs/usage/machine.md]: Maybe we could rename `kubernetes/machines_sets_deployments` to `kubernetes/machine_objects`? does that make it more meaningful?

Comment (zanetworker) [reply docs/usage/machine.md]: Yes it makes sense, I just thought that for ease of access we include the type of the objects in the name but it turned out to be a long one.

Comment (prashanth26) [reply docs/usage/machine.md]: Thanks for the changes.

File: docs/proposals/excess_reserve_capacity.md [hunk 2]
```diff
@@ -1,55 +1,74 @@
 # Excess Reserve Capacity - WIP
```
Comment (hardikdr) [inline docs/proposals/excess_reserve_capacity.md]: This is not WIP anymore, I forgot to remove it later. Can you please use this PR remove `WIP` tag.

File: docs/usage/machine_set.md [hunk 3]
```diff
@@ -1,54 +1,77 @@
 # Maintaining machine replicas using machines-sets
+<!-- TOC -->
 
+- [Maintaining machine replicas using machines-sets](#maintaining-machine-replicas-using-machines-sets)
+  - [Setting up your usage environment](#setting-up-your-usage-environment)
+  - [Important :warning](#important-warning)
```
Comment (hardikdr) [inline docs/usage/machine_set.md]: Is this a typo: warning is not resolved to the symbol

File: docs/usage/machine_set.md [hunk 4]
```diff
@@ -1,54 +1,77 @@
...
 * Follow the [steps described here](prerequisite.md)
 
-## Important :warning: 
-- Make sure that the `kubernetes/machine-set.yaml` points to the same class name as the `kubernetes/aws-machine-class.yaml`.
-- Similarily `kubernetes/aws-machine-class.yaml` secret name and namespace should be same as that mentioned in `kubernetes/aws-secret.yaml`
+## Important :warning
+
+> Make sure that the `kubernetes/machines_sets_deployments/machine-set.yaml` points to the same class name as the `kubernetes/machine_classes/aws-machine-class.yaml`.
+
+> Similarily `kubernetes/machine_classes/aws-machine-class.yaml` secret name and namespace should be same as that mentioned in `kubernetes/secrets/aws-secret.yaml`
```
Comment (hardikdr) [inline docs/usage/machine_set.md]: Name of the actual folder is `Secrets` - can we change it to `secrets`

File: docs/usage/machine_set.md [hunk 5]
```diff
@@ -1,54 +1,77 @@
...
 
 ## Creating machine-set
 
-- Modify `kubernetes/machine-set.yaml` as per your requirement. You can modify the number of replicas to the desired number of machines. Then, create an machine-set
+- Modify `kubernetes/machines_sets_deployments/machine-set.yaml` as per your requirement. You can modify the number of replicas to the desired number of machines. Then, create an machine-set:
+
 ```bash
-$ kubectl apply -f kubernetes/machine-set.yaml
+$ kubectl apply -f kubernetes/machines_sets_deployments/machine-set.yaml
```
Comment (hardikdr) [inline docs/usage/machine_set.md]: I can still see occurrences of `machines_sets_`- which I guess is not aligned with actual folder-structure. Can you please change the necessary occurrences.

File: docs/usage/prerequisite.md [hunk 6]
```diff
@@ -1,9 +1,21 @@
...
+	- [Important :warning](#important-warning)
+	- [Set KUBECONFIG](#set-kubeconfig)
+	- [Replace provider credentials and desired VM configurations](#replace-provider-credentials-and-desired-vm-configurations)
+	- [Deploy required CRDs and Objects](#deploy-required-crds-and-objects)
+	- [Check current cluster state](#check-current-cluster-state)
 
-- Run the Machine Controller Manager either as described in [Setting up a local development environment](../development/local_setup.md) or [Deploying the Machine Controller Manager into a Kubernetes cluster](../deployment/kubernetes.md).
-- Make sure that the following steps are run before managing machines/ machine-sets/ machine-deploys.
+<!-- /TOC -->
+## Important :warning
```
Comment (hardikdr) [inline docs/usage/prerequisite.md]: Should it be resolved to the warning symbol ?

**Pull request merged**